File: android/app/src/main/java/com/wishpool/app/core/asr/AsrManager.kt
package com.wishpool.app.core.asr

import kotlinx.coroutines.flow.StateFlow

/**
 * ASR (Automatic Speech Recognition) 管理器接口
 * 提供语音识别的核心功能
 */
interface AsrManager {
    val state: StateFlow<AsrState>

    suspend fun startRecording()

    suspend fun stopRecording()

    suspend fun reset()

    fun warmUp()
}

File: ios/Sources/WishpoolCore/ASRManager.swift
import Combine
import Foundation

/// ASR 管理器协议，与 Android 端 AsrManager 保持一致。
/// 业务层只依赖该协议，不感知底层引擎细节。
public protocol ASRManager: ObservableObject {
    /// 当前 ASR 状态
    var state: ASRState { get }

    /// ASR 状态发布器（SwiftUI 绑定）
    var statePublisher: Published<ASRState>.Publisher { get }

    /// 开始录音
    func startRecording() async

    /// 停止录音
    func stopRecording() async

    /// 重置会话
    func reset() async

    /// 引擎预热
    func warmUp()
}
